tools/verify_images.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility script to verify that all images have alt text"""
+
+import multiprocessing
+import sys
+import glob
+
+# Dictionary to allowlist lines of code that the checker will not error
+# Format: {"file_path": [list_of_line_numbers]}
+ALLOWLIST_MISSING_ALT_TEXT = {
+    "qiskit_ibm_runtime/fake_provider/__init__.py": [34, 55, 63],
+    "qiskit_ibm_runtime/transpiler/passes/scheduling/dynamical_decoupling.py": [56, 88],
+    "qiskit_ibm_runtime/transpiler/passes/scheduling/__init__.py": [
+        51,
+        102,
+        137,
+        152,
+        176,
+        207,
+        226,
+        240,
+        256,
+        274,
+        294,
+        313,
+        328,
+        342,
+        363,
+        382,
+        406,
+    ],
+}
+
+
+def is_image(line) -> bool:
+    return line.strip().startswith(".. image:") or line.strip().startswith(".. plot:")
+
+
+def allowlist(filename, line_num) -> bool:
+    return (
+        filename in ALLOWLIST_MISSING_ALT_TEXT and line_num in ALLOWLIST_MISSING_ALT_TEXT[filename]
+    )
+
+
+def validate_image(file_path) -> tuple[str, list[str]]:
+    """Validate all the images of a single file"""
+    invalid_images: list[str] = []
+
+    with open(file_path, encoding="utf8") as fd:
+        lines = fd.readlines()
```

Comment:
More direct is `Path(file_path).read_text().splitlines()`. You avoid the whole context manager and close out the file sooner.